includes/internal/emails/generators/class-email-generators-abstract.php

```diff
@@ -80,6 +80,30 @@ public function is_email_active() {
 		return $email && 'publish' === $email->post_status;
 	}
 
+	/**
+	 * Add action if email is active.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @internal
+	 *
+	 * @param string   $action        Action name.
+	 * @param callable $callback      Callback.
+	 * @param int      $priority      Priority.
+	 * @param int      $accepted_args Accepted arguments.
+	 */
+	protected function maybe_add_action( $action, $callback, $priority = 10, $accepted_args = 1 ) {
```

Comment:
I haven't had a chance to test it, but in theory this should work really well based on how WP does actions. Nice solution!